py2puml/parser.py

```diff
@@ -132,13 +132,12 @@ def parse_type(
 ):
     definition_type_fqdn = f'{definition_type.__module__}.{definition_type.__name__}'
     if definition_type_fqdn not in domain_items_by_fqdn:
-        if isinstance(definition_type, type):   # process only class object
-            if issubclass(definition_type, Enum):
-                parse_enum_type(definition_type, definition_type_fqdn, domain_items_by_fqdn)
-            elif getattr(definition_type, '_fields', None) is not None:
-                parse_namedtupled_class(definition_type, definition_type_fqdn, domain_items_by_fqdn)
-            else:
-                parse_class_type(definition_type, definition_type_fqdn, root_module_name, domain_items_by_fqdn, domain_relations)
+        if issubclass(definition_type, Enum):
+            parse_enum_type(definition_type, definition_type_fqdn, domain_items_by_fqdn)
+        elif getattr(definition_type, '_fields', None) is not None:
+            parse_namedtupled_class(definition_type, definition_type_fqdn, domain_items_by_fqdn)
+        else:
+            parse_class_type(definition_type, definition_type_fqdn, root_module_name, domain_items_by_fqdn, domain_relations)
```

Comment:
Thank you for your inputs! I have just realized that the initial code was already recursive :smile: 

I have pushed two commits:
- non-class imports must be filtered out at the proper place: https://github.com/lucsorel/py2puml/pull/8/commits/1a270aa6d6bb0383edd365be41013e3b0fa86d81. `parse_type` handles types (classes) and should not be given a function
- minor changes (project and poetry configurations, gitignore, upgrade of pytest): https://github.com/lucsorel/py2puml/pull/8/commits/816f91c96b16592fbc6504b3b904c871f907dc52

Please, never push [commented-out code like this](https://github.com/lucsorel/py2puml/pull/8/commits/1a270aa6d6bb0383edd365be41013e3b0fa86d81#diff-1f94fc1a3b623c6fbbeaa3b7d3b93c27883ec7aaa96abfbf2c9e93e7dbbba3fc). I believe you did it not to offend me (thanks :smile:) and it could be ok in a 'work-in-progress' commit, but not at the stage of a pull request.

If you agree with these commits, I am ready to merge your contribution. Then I can compile the library and push the new version on pypi.

Reply:
I totally agree with you. Thank you for great work!

Reply:
And one thing more. I added `__init__.py` in `tests` folder to run pytest in vscode.